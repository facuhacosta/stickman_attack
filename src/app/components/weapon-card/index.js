export { WeaponCard } from './weapon-card';