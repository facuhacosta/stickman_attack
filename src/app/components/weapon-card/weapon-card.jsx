import React from 'react';
import style from './weapon-card.module.scss';
  
export function WeaponCard({ weapon }) {

  
  return (
    <div className={style['weapon-card']}>
        <p>{weapon?.name}</p>
        <p>{weapon?.value}</p>
        <p>{weapon?.attack_speed}</p>
        <p>{weapon?.damage}</p>
    </div>  
  );
}
