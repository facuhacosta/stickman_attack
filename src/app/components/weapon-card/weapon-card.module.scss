.weapon-card {
  width: 250px;
  height: 300px;
  border-radius: 10px;
  background-color: gainsboro;
  padding: 12px;
  display: flex;
  flex-direction: column;
  
  .weapon-image {
    background-color: transparent;
    width: 100%;
    height: 50%;
    border-radius: 10px;
    border: solid 1px black;

    & > img {
      width: 100%;
      height: 100%;
      object-fit: contain;
      padding: 5px;
    }
  }

  .weapon-name {
    padding: 4px 0;
    text-align: center;
  }

  .weapon-info {
    border: solid 1px black;
    .weapon-attr {
      display: flex;
      justify-content: space-between;
      border-bottom: 1px black dotted;
      padding: 2px 5px;

      &:nth-last-child(1) {
        border-bottom: none;
      }

      & > p:nth-last-child(1) {
        width: 50px;
        text-align: right;
        background-color: grey;
      }
    }
  }

  .weapon-button {
    background-color: greenyellow;
    width: 150px;
    height: 30px;
    margin: auto;
    border-radius: 5px;
    transition: all 0.1s linear;
    border: none;
    font-weight: 900;
    margin-top: 10px;

    &:hover {
      transform: scale(1.05);
      background-color: rgb(106, 211, 106);
      transition: all 0.1s linear;
      font-weight: bolder;
    }
  }
}