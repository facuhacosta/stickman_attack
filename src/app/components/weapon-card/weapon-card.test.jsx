import React from 'react';
import { render } from '@testing-library/react';
import { WeaponCard } from './weapon-card';

describe('WeaponCard Component', () => {

  // Matching Snapshots
  test('it should match the snapshot', () => {
    const { asFragment } = render(<WeaponCard />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  // Unit tests

  test('it should match the snapshot 2', () => {
    const weapon = {
      name: "pistol",
      value: 5,
      attack_speed: 10,
      damage: 100,
    }
    const { asFragment } = render(<WeaponCard weapon={weapon}/>);
    expect(asFragment()).toMatchSnapshot();
  });

});