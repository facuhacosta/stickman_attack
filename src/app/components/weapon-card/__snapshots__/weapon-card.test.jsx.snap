// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeaponCard Component it should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <p />
    <p />
    <p />
    <p />
  </div>
</DocumentFragment>
`;

exports[`WeaponCard Component it should match the snapshot 2 1`] = `
<DocumentFragment>
  <div>
    <p>
      pistol
    </p>
    <p>
      5
    </p>
    <p>
      10
    </p>
    <p>
      100
    </p>
  </div>
</DocumentFragment>
`;
